packages/main/src/AvatarGroup.js

```diff
@@ -0,0 +1,456 @@
+ * and are not currently visible.
+ *
+ * <h3>Usage</h3>
+ *
+ * Use the <code>AvatarGroup</code> if:
+ * <ul>
+ * <li>You want to display a group of avatars.</li>
+ * <li>You want to display several avatars which have something in common.</li>
+ * </ul>
+ *
+ * Do not use the <code>AvatarGroup</code> if:
+ * <ul>
+ * <li>You want to display a single avatar.</li>
+ * <li>You want to display a gallery for simple images.</li>
+ * <li>You want to use it for other visual content than avatars.</li>
+ * </ul>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.AvatarGroup
+ * @extends UI5Element
+ * @tagname ui5-avatar-group
+ * @since 1.0.0-rc.11
+ * @public
+ */
+class AvatarGroup extends UI5Element {
+	constructor() {
+		super();
+		this._itemNavigation = new ItemNavigation(this);
+
+		this._colorIndex = 0;
+		this._hiddenItems = 0;
+		this._onResizeHandler = this._onResize.bind(this);
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get template() {
+		return AvatarGroupTemplate;
+	}
+
+	static get styles() {
+		return AvatarGroupCss;
+	}
+
+	/**
+	 * Returns array with hidden items. Items that are in overflow button.
+	 * @readonly
+	 * @type { Array }
+	 * @defaultValue []
+	 * @public
+	 */
+	get hiddenItems() {
+		return this.items.slice(this._hiddenStartIndex);
+	}
+
+	get _hiddenStartIndex() {
+		return this._itemsCount - this._hiddenItems;
+	}
+
+	get _overflowBtnHidden() {
+		return this._hiddenItems === 0;
+	}
+
+	get _isGroup() {
+		return this.type === AvatarGroupType.Group;
+	}
+
+	get _itemsCount() {
+		return this.items.length;
+	}
+
+	get _groupTabIndex() {
+		return this._isGroup ? "0" : "-1";
+	}
+
+	get _overflowButtonTabIndex() {
+		return this._isGroup ? "-1" : false;
+	}
+
+	get _colorBackgroundIndex() {
+		// counter is to automatically assign background colors to avatars, `Accent10` is the highest color value
+		if (++this._colorIndex > 10) {
+			this._colorIndex = 1;
+		}
+		return this._colorIndex;
+	}
+
+	/**
+	 * Return the effective overflow button width
+	 * Differences are that when in "Group" type the button is offset and overlaps the avatars
+	 *
+	 * 1) In case of "Group", (LTR/RTL aware) button width is qual to second item offset left/right
+	 * 2) In case of "Individual" group type width is directly taken from button element
+	 * @private
+	 */
+	get _overflowButtonEffectiveWidth() {
+		// if in "Group" mode overflow button size is equal to the offset from second item
+		if (this._isGroup) {
+			let item = this.items[1];
+
+			// in some cases when second avatar is overflowed the offset of the button is the right one
+			if (!item || item.hidden) {
+				item = this._overflowButton;
+			}
+
+			return this.effectiveDir === "rtl" ? this._getWidthToItem(item) : item.offsetLeft;
+		}
+
+		return this._overflowButton.offsetWidth;
+	}
+
+	onAfterRendering() {
+		if (!this._initialRendering) {
```

Comment:
The semantics of the property are wrong. Either call it "_domRefsInitialized" because this is what this boolean is used for, or completely remove it and make "_rootContainer" and "_overflowButton" getters (it will be ok in terms of performance if that's what you mean to optimize)